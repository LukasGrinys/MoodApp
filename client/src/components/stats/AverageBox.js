import React from 'react';
import classNames from 'classnames';
import styles from './AverageBox.module.scss';
import { useTheme } from './../../contexts/ThemeContext';

const AverageBox = ( {children }) => {
    const darkTheme = useTheme();

    return (
        <div className={classNames(
            styles.averageBox,
            darkTheme && styles.dark
        )}>
            {children}
        </div>
    )
}

export default AverageBox;