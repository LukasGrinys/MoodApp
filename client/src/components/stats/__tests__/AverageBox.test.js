import React from 'react';
import AverageBox from '../AverageBox';
import { shallow } from 'enzyme';

const testAverage = 10;
const wrapper = shallow(<AverageBox>{testAverage}</AverageBox>)

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

/* Indication, that average mood rating 
should be calculated inside this component */
test('Renders the average passed as props', () => {
    expect(wrapper.text()).toBe(testAverage.toString());
});