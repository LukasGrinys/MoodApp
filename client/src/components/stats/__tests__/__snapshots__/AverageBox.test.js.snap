// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="averageBox"
>
  10
</div>
`;
